osdconfig/api.go

```diff
@@ -96,26 +69,37 @@ func (manager *configManager) error() error {
 // Close closes the contexts and releases bookkeeping memory
 func (manager *configManager) Close() {
 	// cancel contexts and wait for any pending tasks to complete
+	logrus.WithField("source", sourceManagerClose).
+		Info(msgCtxCancelled)
 	manager.runCancel()
 	manager.cancel()
 	manager.wait()
 
+	// introduce small delay to allow associated processes linked via
+	// contexts to respond to context cancellations
+	time.Sleep(time.Second)
+
 	// clean up resources
+	logrus.WithField("source", sourceManagerClose).
+		Info(msgMemRelease)
 	manager.cc = nil
 	manager.cb = nil
-	manager.opt = nil
+	logrus.WithField("source", sourceManagerClose).
+		Info(msgCleanup)
 }
 
-// Abort sends context cancellation to running callbacks if any
+// abort sends context cancellation to running callbacks if any
 func (manager *configManager) abort() {
-	logrus.Info("aborting via context cancellation")
+	logrus.WithField("source", sourceManagerAbort).
+		Warn(msgAborting)
 	manager.runCancel()
 	manager.Lock()
```

Comment:
Why lock here only to log?

Reply:
lock was there to sync with other threads that release lock on completion.